src/tasks/minions/buyActivity.ts

```diff
@@ -0,0 +1,53 @@
+import { Bank, toKMB } from 'oldschooljs';
+
+import { Time } from 'e';
+import Buyables from '../../lib/data/buyables/buyables';
+import type { BuyActivityTaskOptions } from '../../lib/types/minions';
+import { itemNameFromID } from '../../lib/util';
+import { handleTripFinish } from '../../lib/util/handleTripFinish';
+import { updateBankSetting } from '../../lib/util/updateBankSetting';
+
+export const buyTask: MinionTask = {
+	type: 'Buy',
+	async run(data: BuyActivityTaskOptions) {
+		const { userID, channelID, itemID, quantity } = data;
+		const user = await mUserFetch(userID);
+
+		const buyable = Buyables.find(b => {
+			if (!b.quantityPerHour || !b.outputItems) return false;
+
+			const bank = typeof b.outputItems === 'function' ? b.outputItems(user) : b.outputItems;
+
+			return bank.has(itemID);
```

Comment:
This is a little sketchy but seems okay, maybe also a bank.length === 1 just to add some extra safety